pvlib/bifacial/infinite_sheds.py

```diff
@@ -0,0 +1,792 @@
+    #  :     \\.___________\
+    #  :       \  ^*-.  psi \
+    #  :        \  x   *-.   \
+    #  :         \  v      *-.\
+    #  :          \<-----P---->\
+
+    x1 = x * sind(surface_tilt)
+    x2 = (x * cosd(surface_tilt) + 1 / gcr)
+    psi = np.arctan2(x1, x2)  # do this first because it handles 0 / 0
+    return np.rad2deg(psi)
+
+
+def _vf_row_ground(x, surface_tilt, gcr):
+    """
+    View factor from a point x on the row to the ground.
+
+    Parameters
+    ----------
+    x : numeric
+        Fraction of row slant height from the bottom. [unitless]
+    surface_tilt : numeric
+        Surface tilt angle in degrees from horizontal, e.g., surface facing up
+        = 0, surface facing horizon = 90. [degree]
+    gcr : float
+        Ground coverage ratio, ratio of row slant length to row spacing.
+        [unitless]
+
+    Returns
+    -------
+    vf : numeric
+        View factor from the point at x to the ground. [unitless]
+
+    """
+    cst = cosd(surface_tilt)
+    # angle from horizontal at the point x on the row slant height to the
+    # bottom of the facing row
+    psi_t_shaded = _ground_angle(x, surface_tilt, gcr)
+    # view factor from the point on the row to the ground
+    return 0.5 * (cosd(psi_t_shaded) - cst)
+
+
+def _vf_row_ground_integ(f_x, surface_tilt, gcr, npoints=100):
+    """
+    View factors to the ground from shaded and unshaded parts of a row.
+
+    Parameters
+    ----------
+    f_x : numeric
+        Fraction of row slant height from the bottom that is shaded. [unitless]
+    surface_tilt : numeric
+        Surface tilt angle in degrees from horizontal, e.g., surface facing up
+        = 0, surface facing horizon = 90. [degree]
+    gcr : float
+        Ground coverage ratio, ratio of row slant length to row spacing.
+        [unitless]
+    npoints : int, default 100
+        Number of points for integration. [unitless]
+
+    Returns
+    -------
+    vf_shade_ground_integ : numeric
+        View factor from the shaded portion of the row to the ground.
+        [unitless]
+    vf_noshade_ground_integ : numeric
+        View factor from the unshaded portion of the row to the ground.
+        [unitless]
+
+    Notes
+    -----
+    The view factor to the ground at a point x along the row slant height is
+    given by
+
+    .. math ::
+        \\large{f_{gr} = \frac{1}{2} \\left(\\cos\\left(\\psi_t\\right) -
+        \\cos \\left(\\beta\\right) \\right)
+
+    where :math:`\\psi_t` is the angle from horizontal of the line from point
+    x to the bottom of the facing row, and :math:`\\beta` is the surface tilt.
+
+    Each view factor is integrated over the relevant portion of the row
+    slant height.
+    """
+    # handle Series inputs
+    surface_tilt = np.array(surface_tilt)
+    # shaded portion of row slant height
+    x = np.linspace(0 * f_x, f_x, num=npoints)
+    # view factor from the point on the row to the ground
+    y = _vf_row_ground(gcr, surface_tilt, x)
```

Comment:
@cwhanse there is a bug here and on line 417.  The parameter list for `_vf_row_ground` is `x, surface_tilt, gcr`.